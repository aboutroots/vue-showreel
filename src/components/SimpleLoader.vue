<template>
  <div class="spinner-wrapper" :class="{ fullPage }">
    <ios-sync-icon class="spinner" w="40" h="40" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'SimpleLoader',
  props: {
    fullPage: {
      type: Boolean,
    },
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';

.spinner-wrapper {
  width: 100%;
  padding: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;

  &.fullPage {
    margin-top: -4rem;
    height: 100vh;
  }
}
.spinner {
  animation: spin 1000ms linear infinite;
  color: $primary;
}
@keyframes spin {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
</style>
