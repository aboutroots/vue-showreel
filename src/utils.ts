export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
