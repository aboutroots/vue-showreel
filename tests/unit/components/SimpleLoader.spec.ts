import SimpleLoader from '@/components/SimpleLoader.vue';
import { shallowMount } from '../utils';

describe('SimpleLoader tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(SimpleLoader);

    const container = wrapper.find('.SimpleLoader');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(SimpleLoader);
    expect(wrapper.element).toMatchSnapshot();
  });
});
