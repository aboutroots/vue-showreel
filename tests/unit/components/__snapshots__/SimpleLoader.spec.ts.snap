// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleLoader tests suite matches snapshot 1`] = `
<div
  class="SimpleLoader spinner-wrapper"
>
  <ios-sync-icon-stub
    animate=""
    class="spinner"
    h="40"
    rootclass=""
    title=""
    w="40"
  />
</div>
`;
